infra/aws/terraform/prow-build-cluster/eks.tf

```diff
@@ -0,0 +1,120 @@
+  version = "19.10.0"
+
+  # General cluster properties.
+  cluster_name                   = var.cluster_name
+  cluster_version                = var.cluster_version
+  cluster_endpoint_public_access = true
+
+  # Manage aws-auth ConfigMap.
+  manage_aws_auth_configmap = true
+
+  # We do not really care if we use IPv4 or IPv6 as we don't depend much on
+  # pod-to-pod networking.
+  cluster_ip_family = "ipv4"
+
+  vpc_id                   = module.vpc.vpc_id
+  subnet_ids               = module.vpc.private_subnets
+  control_plane_subnet_ids = module.vpc.intra_subnets
+
+  cluster_addons = {
+    coredns = {
+      most_recent = true
+    }
+    kube-proxy = {
+      most_recent = true
+    }
+    vpc-cni = {
+      most_recent              = true
+      service_account_role_arn = module.vpc_cni_irsa.iam_role_arn
+    }
+    aws-ebs-csi-driver = {
+      most_recent              = true
+      service_account_role_arn = module.ebs_csi_irsa.iam_role_arn
+    }
+  }
+
+  eks_managed_node_group_defaults = {
+    ami_id                     = var.node_ami
+    enable_bootstrap_user_data = true
+    instance_types             = var.node_instance_types
+
+    # We are using the IRSA created below for permissions
+    # However, we have to deploy with the policy attached FIRST (when creating a fresh cluster)
+    # and then turn this off after the cluster/node group is created. Without this initial policy,
+    # the VPC CNI fails to assign IPs and nodes cannot join the cluster
+    # See https://github.com/aws/containers-roadmap/issues/1666 for more context
+    iam_role_attach_cni_policy = false
+  }
+
+  eks_managed_node_groups = {
+    # Build cluster node group.
+    build = {
+      name            = "build-managed"
+      description     = "EKS managed node group used for build nodes"
+      use_name_prefix = true
+
+      subnet_ids = module.vpc.private_subnets
+
+      min_size     = var.node_min_size
+      max_size     = var.node_max_size
+      desired_size = var.node_desired_size
+
+      ami_id                     = var.node_ami
+      enable_bootstrap_user_data = true
+
+      # Force version update if existing pods are unable to be drained due to a PodDisruptionBudget issue.
+      force_update_version = true
+      update_config = {
+        max_unavailable_percentage = 33
+      }
+
+      # Required to ensure Prow works well.
+      pre_bootstrap_user_data = <<-EOT
+        sysctl -w fs.inotify.max_user_watches=524288
+      EOT
+
+      capacity_type  = "ON_DEMAND"
+      instance_types = var.node_instance_types
+
+      ebs_optimized     = true
+      enable_monitoring = true
+
+      block_device_mappings = {
+        xvda = {
+          device_name = "/dev/xvda"
+          ebs = {
+            volume_size           = var.node_volume_size
+            volume_type           = "gp3"
+            encrypted             = true
+            kms_key_id            = module.ebs_kms_key.key_arn
+            delete_on_termination = true
```

Comment:
We should get the max of IOPS and throughput for the EBS disks. we currently are over 11k of IOPS for the existing one.